tools/toolchain/scripts/install_cosma.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash -e
+[ "${BASH_SOURCE[0]}" ] && SCRIPT_NAME="${BASH_SOURCE[0]}" || SCRIPT_NAME=$0
+SCRIPT_DIR="$(cd "$(dirname "$SCRIPT_NAME")" && pwd -P)"
+
+cosma_ver="1.0.0"
+cosma_sha256="c142104258dcca4c17fa7faffc2990a08d2777235c7980006e93c5dca51061f6"
+source "${SCRIPT_DIR}"/common_vars.sh
+source "${SCRIPT_DIR}"/tool_kit.sh
+source "${SCRIPT_DIR}"/signal_trap.sh
+source "${INSTALLDIR}"/toolchain.conf
+source "${INSTALLDIR}"/toolchain.env
+
+[ -f "${BUILDDIR}/setup_cosma" ] && rm "${BUILDDIR}/setup_cosma"
+
+! [ -d "${BUILDDIR}" ] && mkdir -p "${BUILDDIR}"
+cd "${BUILDDIR}"
+
+case "$with_cosma" in
+    __INSTALL__)
+        echo "==================== Installing cosma ===================="
+        pkg_install_dir="${INSTALLDIR}/cosma-${cosma_ver}"
+        install_lock_file="$pkg_install_dir/install_successful"
+        if verify_checksums "${install_lock_file}" ; then
+            echo "cosma-${cosma_ver} is already installed, skipping it."
+        else
+            if [ -f cosma-${cosma_ver}.tar.gz ] ; then
+                echo "cosma-${cosma_ver}.tar.gz is found"
+            else
+                download_pkg ${DOWNLOADER_FLAGS} ${cosma_sha256} \
+                             "https://github.com/eth-cscs/COSMA/releases/download/1.0/cosma.tar.gz" \
+                              -o cosma-${cosma_ver}.tar.gz
+
+            fi
+            echo "Installing from scratch into ${pkg_install_dir}"
+            [ -d cosma-${cosma_ver} ] && rm -rf cosma-${cosma_ver}
+            tar -xzf cosma-${cosma_ver}.tar.gz
+            mv cosma cosma-${cosma_ver}
+            cd cosma-${cosma_ver}
+            mkdir build-cpu
+            cd build-cpu
+            case "$FAST_MATH_MODE" in
+                mkl)
+                cmake -DCMAKE_INSTALL_PREFIX="${pkg_install_dir}" \
+                          -DCOSMA_WITH_MKL_BLAS=ON \
+                          -DCOSMA_WITH_MKL_ScaLAPACK=ON .. \
+                          -DCOSMA_WITH_INSTALL=ON > cmake.log 2>&1
```

Comment:
please either ensure that COSMA is using `lib` as libdir on all systems or adjust the paths referencing `lib/` in this script to also work with `lib64` instead, see other install scripts for similar issues